changelogs/unreleased/4384-regular-timeouts-on-core-tests.yml

```
@@ -0,0 +1,6 @@
+description: Fix deadlock in the database locking mechanism.
```

Let's make the changelog message a bit less alarming:
```suggestion
description: Fix rare deadlock in the database locking mechanism when tasks are cancelled, mostly affects test environments.
```